qiita_pet/templates/redbiom.html

```diff
@@ -0,0 +1,192 @@
+{% from qiita_core.qiita_settings import qiita_config %}
+{% from future.utils import viewitems %}
+{% extends sitebase.html %}
+
+{%block head%}
+<script type="text/javascript" src="{% raw qiita_config.portal_dir %}/static/js/sharing.js"></script>
+<script type="text/javascript">
+  $(document).ready(function() {
+    qiita_websocket.init(window.location.host + '{% raw qiita_config.portal_dir %}/study/list/socket/', error, error);
+    qiita_websocket.add_callback('sel', show_alert);
+    $("#redbiom-table").dataTable({
+      "dom": '<"ps">lfr<t><"footer">ip',
+      'footerCallback': function ( row, data, start, end, display ) {
+
+        // adding total features found
+
+        var api = this.api(), data;
+        var total_features = api
+          .column(0)
+          .data()
+          .reduce( function (a, b) {return a + b.length}, 0);
+
+        $('.footer').addClass("col-md-12 text-right");
+        if (total_features == 0) {
+          text = '';
+        } else {
+            text = 'Found ' + total_features + ' artifacts.';
+        }
+        $('.footer').html(text)
+      },
+      "columns": [
+        { "orderable": false,       "width": "15%", "data": "artifact_biom_ids"},
+        { "data": "study_title",    "width": "70%" },
+        { "data": "study_abstract", "width": "0%" },
+        { "data": "study_id",       "width": "15%" },
+        { "data": "study_alias",    "width": "0%" }],
+      columnDefs: [
+        // {type:'natural', targets:[2,6,7]},
+        {"targets": [ 2, 4 ], "visible": false},
+        // render zero
+        {"render": function ( data, type, row, meta ) {
+          if (data !== null && data !== undefined && data.length != 0){
+            {% if current_user is not None %}
+              return '<div class="container" style="max-width: 5em;">'+
+                       '<div class="row justify-content-md-center">' +
+                         '<div class="col-md-1 text-center details-control">&nbsp;</div>' +
```

Comment:
It looks like these two blocks are the same except for this line? You can then do:
```
              return '<div class="container" style="max-width: 5em;">'+
                       '<div class="row justify-content-md-center">' +
            {% if current_user is not None %}
                         '<div class="col-md-1 text-center details-control">&nbsp;</div>' +
            {% end %}
                         '<div class="col-md-1 text-center">' + data.length + '</div>' +
                       '</div>' +
                     '</div>';
```

Reply:
k